packages/popconfirm/src/index.vue

```diff
@@ -0,0 +1,103 @@
+<template>
+  <el-popper trigger="click" manualMode :value="visible" effect="light">
+    <template #default>
+    <div class="el-popconfirm">
+      <p class="el-popconfirm__main">
+        <i
+          v-if="!hideIcon"
+          :class="icon"
+          class="el-popconfirm__icon"
+          :style="{color: iconColor}"
+        ></i>
+        {{title}}
+      </p>
+      <div class="el-popconfirm__action">
+        <el-button
+          size="mini"
+          :type="cancelButtonType"
+          @click="cancel"
+        >
+          {{cancelButtonText}}
+        </el-button>
+        <el-button
+          size="mini"
+          :type="confirmButtonType"
+          @click="confirm"
+        >
+          {{confirmButtonText}}
+        </el-button>
+      </div>
+    </div>
+    </template>
+    <template #trigger>
+      <div style="display: inline-block" @click="open">
+        <slot name="reference"></slot>
+      </div>
+    </template>
+  </el-popper>
+</template>
+
+<script lang="ts">
+  import { Popper as ElPopper } from '@element-plus/popper'
+  import { defineComponent,ref } from 'vue';
+  export default defineComponent({
+    name: 'ElPopconfirm',
+    props: {
+      title: {
+        type: String
+      },
+      confirmButtonText: {
+        type: String,
+        default: '确定'
+      },
+      cancelButtonText: {
+        type: String,
+        default: '取消'
+      },
+      confirmButtonType: {
+        type: String,
+        default: 'primary'
+      },
+      cancelButtonType: {
+        type: String,
+        default: 'text'
+      },
+      icon: {
+        type: String,
+        default: 'el-icon-question'
+      },
+      iconColor: {
+        type: String,
+        default: '#f90'
+      },
+      hideIcon: {
+        type: Boolean,
+        default: false
+      }
+    },
+    components: {
+      ElPopper,
+    },
+    setup(props,ctx){
+      const visible = ref(false)
+
+      const confirm = (): void => {
```

Comment:
`void` is not need.

Same for the following.

Reply:
Correct, typescript automatically infers the return type as `void`, it is redundant.
BTW "is not need(ed)"  a passive statement here.

Reply:
Ok, I will modify this